vulnerabilities/tests/test_pysec.py

```diff
@@ -0,0 +1,292 @@
+                                "3.1.0.rc1",
+                                "3.1.0rc1",
+                                "3.1.1",
+                                "3.1.2",
+                                "3.2.0",
+                                "3.3.0",
+                                "3.3.1",
+                                "3.3.2",
+                                "3.3.3",
+                                "3.4.0",
+                                "3.4.1",
+                                "3.4.2",
+                                "4.0.0",
+                                "4.1.0",
+                                "4.1.1",
+                                "4.2.0",
+                                "4.2.1",
+                                "4.3.0",
+                                "5.0.0",
+                                "5.1.0",
+                                "5.2.0",
+                                "5.3.0",
+                                "5.4.0",
+                                "5.4.0.dev0",
+                                "5.4.1",
+                                "6.0.0",
+                                "6.1.0",
+                                "6.2.0",
+                                "6.2.1",
+                                "6.2.2",
+                                "7.0.0",
+                                "7.1.0",
+                                "7.1.1",
+                                "7.1.2",
+                                "7.2.0",
+                                "8.0.0",
+                                "8.0.1",
+                                "8.1.0",
+                                "8.1.1",
+                                "8.1.2",
+                                "8.2.0",
+                                "8.3.0",
+                                "8.3.1",
+                                "8.3.2",
+                                "8.4.0",
+                            ]
+                        ),
+                        fixed_version=PypiVersion("9.0.0"),
+                    )
+                ],
+                references=[
+                    Reference(
+                        url="https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#restrict-builtins"
+                        "-available-to-imagemath-eval",
+                        severities=[],
+                    ),
+                    Reference(
+                        url="https://lists.debian.org/debian-lts-announce/2022/01/msg00018.html",
+                        severities=[],
+                    ),
+                    Reference(
+                        url="https://github.com/advisories/GHSA-8vj2-vxx3-667w", severities=[]
+                    ),
+                ],
+                date_published=datetime.datetime(2022, 1, 10, 14, 12, 0, 0).replace(
+                    tzinfo=datetime.timezone.utc
+                ),
+            ),
+        ]
+        found_data = []
+        for response in self.mock_response:
```

Comment:
Why do you use a mock response?
This may be OK for integration testing, but we need unit tests.
You could have a much simpler testing by testing simply one item at a time in parse_advisory_data() and have multiple test functions, one for each specific case that needs to be checked